lib/castle_web/search.ex

```diff
@@ -0,0 +1,47 @@
+defmodule CastleWeb.Search do
+
+  import Ecto.Query
+
+  def prefix_search(query) do
+
+    ends_with_whitespace = Regex.match?(~r/\s$/, query)
+    query = String.trim(query)
+
+    query = query
+            |> String.split
+            |> Enum.map(fn term ->
+              # strip all special characters from term
+              term
+              |> String.replace( ~r/[\W]+/u, "")
```

Comment:
The `tsquery` is expecting well formed syntax, so we have to clean up user input to remove characters that reflect on tsquery semantics. Removing all special characters here fixes all the problems I encountered.

The alternative is to drop prefix search entirely and use `plainto_tsquery` down in line 37.

Reply:
This works for me.  Did something similar for the simple-text-elasticsearch in CMS.